src/components/AnalyticsTrends.tsx

```
@@ -0,0 +1,337 @@
+    const weekly = Object.keys(weeklyMap).map(week => ({
+      label: week,
+      value: Math.round(weeklyMap[week].sum / weeklyMap[week].count),
+    })).slice(-4); // Keep last 4 weeks
+
+    // 3. Vendor (market) aggregates
+    const vendorMap: Record<string, { sum: number; count: number; previousSum: number; previousCount: number }> = {};
+    scans.forEach(s => {
+      const market = s.market_name || 'General Wet Stall';
+      if (!vendorMap[market]) vendorMap[market] = { sum: 0, count: 0, previousSum: 0, previousCount: 0 };
+      
+      const isRecent = new Date(s.timestamp).getTime() > Date.now() - 7 * 24 * 60 * 60 * 1000;
```

ESLint

[error] Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

/home/jailuser/git/src/components/AnalyticsTrends.tsx:60:58
  58 |       if (!vendorMap[market]) vendorMap[market] = { sum: 0, count: 0, previousSum: 0, previousCount: 0 };
  59 |       
> 60 |       const isRecent = new Date(s.timestamp).getTime() > Date.now() - 7 * 24 * 60 * 60 * 1000;
     |                                                          ^^^^^^^^^^ Cannot call impure function
  61 |       if (isRecent) {
  62 |         vendorMap[market].sum += s.freshness_index;
  63 |         vendorMap[market].count += 1;

(react-hooks/purity)